hastory-data/src/Data/Hastory/API.hs

```diff
@@ -51,32 +46,25 @@ instance ToHttpApiData Token where
   toUrlPiece (Token token) = toUrlPiece token
 
 -- | Main Hastory API specification.
-type HastoryAPI = "commands" :> "append" :> Header TokenHeaderKey Token :> ReqBody '[JSON] EntryWithKey :> Post '[JSON] ()
+type HastoryAPI
+   = "commands" :> "append" :> Header TokenHeaderKey Token :> ReqBody '[ JSON] Entry :> Post '[ JSON] ()
 
 api :: Proxy HastoryAPI
 api = Proxy
 
-data HastoryClient = HastoryClient ClientEnv Token
-
-instance Show HastoryClient where
-  show (HastoryClient (ClientEnv _ baseUrl _) _) = "HastoryClient with baseUrl " <> show baseUrl
-
--- | By default parseBaseUrl throws exceptions and we don't like exceptions here.
--- Therefore we convert it to an action in IO and MonadError T.Text context.
-parseBaseUrlSafe :: (MonadThrow m, MonadBaseControl IO m, MonadError T.Text m) => T.Text -> m BaseUrl
-parseBaseUrlSafe url =
-  parseBaseUrl (T.unpack url)
-    `catch` (\(_ :: InvalidBaseUrlException) -> throwError "Base url couldn't be parsed")
+data HastoryClient =
+  HastoryClient ClientEnv Token
 
 -- * Hastory Client
-
 -- | Creates a hastory client type.
 --
 -- This type is needed because creating & destroying HTTP managers are expensive.
 -- Once a user gets a HastoryClient, it's being used throughout the entire life of the user.
-mkHastoryClient :: (MonadError T.Text m, MonadThrow m, MonadBaseControl IO m, MonadIO m) => T.Text -> Token -> m HastoryClient
+mkHastoryClient ::
+     (MonadError T.Text m, MonadIO m) => T.Text -> Token -> m HastoryClient
 mkHastoryClient baseUrl token = do
```

Comment:
It may be better to just pass in the `BaseUrl` into this function.
`MonadError` isn't very nice usually.
To get a `BaseUrl`, use 
```case parseBaseUrl s of
  Nothing -> die $ "invalid url: " <> s
  Just burl -> ...```

There's also need to make the thing that we parse a `Text` if it has to be unpacked again afterward.

Reply:
Using `die` would also generate an error from the client side, which I understand as something we try to avoid. I'd like to keep it as is and log a warning in the case of an error, including base url not being able to be parsed.

Reply:
A warning instead of an error is fine, but that still doesn't mean that you have to throw it _here_ in the code.